azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java

```diff
@@ -71,8 +71,8 @@ public static String encodeFieldsToUniqueIdCustom(VMImage imageReference) {
    }
 
    public static VMImage decodeFieldsFromUniqueId(final String id) {
-      String[] fields = checkNotNull(id, "id").split("/");
       VMImage vmImage;
+      String[] fields = checkNotNull(id, "id").split("/");
       boolean custom = fields.length == 5;
```

Comment:
Change this to a public static function that can be called from the adapter, instead of duplicating this logic there.